Make `cs3110 compile` learn the `-use-menhir` flag

remyjette opened the pull request:
If `-use-menhir` is passed to cs3110 compile, `-use-menhir` will be set as a flag for ocamlbuild.

remyjette commented:
Maybe remove the `-menhir` alias, since that's a *different* flag for ocamlbuild.

Before merging, should modify [cli_config.ml](https://github.com/cs3110/tools/cs3110-cli/util/cli_config.ml) to learn the use-menhir flag so that it can be set in a `.cs3110` file.

mdgeorge4153 commented:
We should probably just show students how to add it to the .cs3110 file,
like we did for async in today's recitation.

-M

On 10/28/2015 12:42 PM, Remy Jette wrote:
>
> Maybe remove the |-menhir| alias, since that's a /different/ flag for
> ocamlbuild.
>
> Before merging, should modify cli_config.ml
> <https://github.com/cs3110/tools/cs3110-cli/util/cli_config.ml> to
> learn the use-menhir flag so that it can be set in a |.cs3110| file.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cs3110/tools/pull/128#issuecomment-151904769>.
>

remyjette commented:
Today's recitation is actually what prompted adding this comment to the PR.

I added this note because as it currently stands, merging this would only allow for the `-use-menhir` flag to be passed to `cs3110 compile`. The .cs3110 file is handled and parsed by cli_config.ml, and right now it contains no references to menhir. As a result we can't show students how to add it to the .cs3110 file, because there's currently nothing to add.

As a result, I've added this note to ensure that cli_config is updated to recognize something like `compile.use_menhir=true` before this PR is merged.

Then, in the future we can tell students to add `compile.use_menhir=true` (or whatever the flag is) to their .cs3110 file for it to be enabled. Just gotta add that flag first.

remyjette commented:
Pushed directly to master as
ccfba83a806898166e2e0b5b939741edce4b5036
097f313747db55a116ffdb6d824ca9d35f843e6f
3e52d4318516bed4535213d93e8a88d93273fee5
